streams/src/test/java/org/apache/kafka/streams/integration/KStreamAggregationDedupIntegrationTest.java

```diff
@@ -314,11 +247,11 @@ private void startStreams() {
             keyDeserializer.getClass().getName());
         consumerProperties.setProperty(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
             valueDeserializer.getClass().getName());
-        return IntegrationTestUtils.waitUntilMinKeyValueRecordsReceived(consumerProperties,
-            outputTopic,
-            numMessages,
-            60 * 1000);
+        consumerProperties.setProperty(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, String.valueOf(Integer.MAX_VALUE));
```

Comment:
I take it this was the source of the flakiness. Can you explain why, for my education?

Reply:
The flakiness is actually that for this dedup integration test, we should check that "for each key, the last record is the expected value", while previously we just check that "we retrieve N records, and check that these N records are exactly the expected values". However even with dedup based on caching, it may not be the case that we ONLY produce N final records.

This PR increases the likelihood that we do not only produce N final records, and hence I updated the check logic accordingly.

Reply:
I think this is not needed, will remove this.